scene/gui/flow_container.cpp

```diff
@@ -0,0 +1,249 @@
+/*                                                                       */
+/* Permission is hereby granted, free of charge, to any person obtaining */
+/* a copy of this software and associated documentation files (the       */
+/* "Software"), to deal in the Software without restriction, including   */
+/* without limitation the rights to use, copy, modify, merge, publish,   */
+/* distribute, sublicense, and/or sell copies of the Software, and to    */
+/* permit persons to whom the Software is furnished to do so, subject to */
+/* the following conditions:                                             */
+/*                                                                       */
+/* The above copyright notice and this permission notice shall be        */
+/* included in all copies or substantial portions of the Software.       */
+/*                                                                       */
+/* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,       */
+/* EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF    */
+/* MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.*/
+/* IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY  */
+/* CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,  */
+/* TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE     */
+/* SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                */
+/*************************************************************************/
+
+#include "flow_container.h"
+
+void FlowContainer::_resort() {
+	int separation_horizontal = get_theme_constant(SNAME("hseparation"));
+	int separation_vertical = get_theme_constant(SNAME("vseparation"));
+
+	bool rtl = is_layout_rtl();
+
+	Vector<Rect2> child_rects;
+
+	// Stores the number of children per line and the calculated line height.
+	Vector<Vector2i> lines_data;
+
+	Vector2 ofs;
+	int line_height = 0;
+	int current_width = vertical ? get_rect().size.y : get_rect().size.x;
+	int children_in_current_line = 0;
+
+	for (int i = 0; i < get_child_count(); i++) {
+		Control *child = Object::cast_to<Control>(get_child(i));
+		if (!child || !child->is_visible_in_tree()) {
+			continue;
+		}
+		if (child->is_set_as_top_level()) {
+			continue;
+		}
+
+		Size2i child_min_size = child->get_combined_minimum_size();
+
+		if (vertical) { /* VERTICAL */
+			if (children_in_current_line > 0) {
+				ofs.y += separation_vertical;
+			}
+			if (ofs.y + child_min_size.y > current_width) {
+				// Move in new column (vertical line).
+				ofs.y = 0;
+				ofs.x += line_height + separation_horizontal;
+				lines_data.push_back(Vector2i(children_in_current_line, line_height));
+
+				line_height = 0;
+				children_in_current_line = 0;
+			}
+
+			child_rects.push_back(Rect2(ofs, child_min_size));
+
+			if (child_min_size.x > line_height) {
+				line_height = child_min_size.x;
+			}
```

Comment:
Could be replaced with `line_height = max(line_height, child_min_size.x)`

Reply:
Replaced this and all similar spots with the MAX(x,y) macro.